tests/v2/validation/fleet/public_repo_test.go

```diff
@@ -0,0 +1,166 @@
+	defer f.session.Cleanup()
+
+	var clusterID string
+	var err error
+	clusterObject, _, _ := extensionscluster.GetProvisioningClusterByName(f.client, f.client.RancherConfig.ClusterName, "fleet-default")
+	if clusterObject != nil {
+		status := &provv1.ClusterStatus{}
+		err := steveV1.ConvertToK8sType(clusterObject.Status, status)
+		require.NoError(f.T(), err)
+
+		clusterID = status.ClusterName
+		logrus.Info(clusterID)
+	} else {
+		clusterID, err = extensionscluster.GetClusterIDByName(f.client, f.client.RancherConfig.ClusterName)
+		require.NoError(f.T(), err)
+	}
+
+	podErrors := pods.StatusPods(f.client, clusterID)
+	require.Empty(f.T(), podErrors)
+
+	fleetVersion, err := fleet.GetDeploymentVersion(f.client, fleet.FleetControllerName, "local")
+	require.NoError(f.T(), err)
+
+	chartVersion, err := f.client.Catalog.GetLatestChartVersion("fleet", catalog.RancherChartRepo)
+	require.NoError(f.T(), err)
+	// fleet chart version may contain chart version info that is a superset of the version reported by the fleet deployment.
+	require.Contains(f.T(), chartVersion, fleetVersion[1:])
+
+	f.Run("/fleet "+fleetVersion, func() {
+		fleetGitRepo := v1alpha1.GitRepo{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "automatedrepo-" + namegenerator.RandStringLower(5),
+				Namespace: "fleet-default",
+			},
+			Spec: v1alpha1.GitRepoSpec{
+				Repo:                "https://github.com/rancher/fleet-examples", // "https://github.com/rancher/fleet-examplestypo",
+				Branch:              "master",
+				Paths:               []string{"simple"},
+				CorrectDrift:        &v1alpha1.CorrectDrift{},
+				ImageScanCommit:     v1alpha1.CommitSpec{AuthorName: "", AuthorEmail: ""},
+				ForceSyncGeneration: 1,
+				Targets:             []v1alpha1.GitTarget{{ClusterName: f.client.RancherConfig.ClusterName}},
+			},
+		}
+
+		f.client, err = f.client.ReLogin()
+		require.NoError(f.T(), err)
+
+		logrus.Info("Deploying public fleet gitRepo")
+		gitRepoObject, err := extensionsfleet.CreateFleetGitRepo(f.client, &fleetGitRepo)
+		require.NoError(f.T(), err)
+
+		fleet.VerifyGitRepo(f.T(), f.client, gitRepoObject.ID, clusterID, "fleet-default/"+f.client.RancherConfig.ClusterName)
+
+	})
+}
+
+func (f *FleetPublicRepoTestSuite) TestDynamicGitRepoDeployment() {
+	defer f.session.Cleanup()
+
+	var clusterID string
+	var err error
+	clusterObject, _, _ := extensionscluster.GetProvisioningClusterByName(f.client, f.client.RancherConfig.ClusterName, "fleet-default")
+	if clusterObject != nil {
+		status := &provv1.ClusterStatus{}
+		err := steveV1.ConvertToK8sType(clusterObject.Status, status)
+		require.NoError(f.T(), err)
+
+		clusterID = status.ClusterName
+		logrus.Info(clusterID)
+	} else {
+		clusterID, err = extensionscluster.GetClusterIDByName(f.client, f.client.RancherConfig.ClusterName)
+		require.NoError(f.T(), err)
+	}
+
+	podErrors := pods.StatusPods(f.client, clusterID)
+	require.Empty(f.T(), podErrors)
+
+	dynamicGitRepo := fleet.GitRepoConfig()
```

Comment:
can we add an error check to make sure GitRepoConfig is not empty otherwise we might get a null pointer panic